climada/util/interpolation.py

```diff
@@ -186,3 +202,95 @@ def index_nn_haversine(centroids, coordinates, threshold=THRESHOLD):
 
     # Copy result to all exposures and return value
     return np.squeeze(assigned[inv])
+
+
+def index_nn_euclidian(centroids, coordinates, threshold=THRESHOLD,
+                       check_antimeridian=True):
+    """Compute the neareast centroid for each coordinate using a k-d tree.
+
+    Parameters
+    ----------
+    centroids : 2d array
+        First column contains latitude, second column contains longitude.
+        Each row is a geographic point
+    coordinates : 2d array
+        First column contains latitude, second column contains longitude. Each
+        row is a geographic point
+    threshold : float
+        distance threshold in km over which no neighbor will be found. Those
+        are assigned with a -1 index
+    check_antimedirian: bool
+        if True, recomputes the nn in a strip with lon size equal to threshold
+        around the antimeridian using the Haversine distance.
+
+    Returns
+    -------
+    np.array
+        with so many rows as coordinates containing the centroids indexes
+    """
+    # Construct tree from centroids
+    tree = sp.spatial.KDTree(np.radians(centroids))
+    # Select unique exposures coordinates
+    _, idx, inv = np.unique(coordinates, axis=0, return_index=True,
+                            return_inverse=True)
+
+    # query the k closest points of the n_points using dual tree
+    dist, assigned = tree.query(np.radians(coordinates[idx]), k=1, p=2, workers=-1)
+
+    # Raise a warning if the minimum distance is greater than the
+    # threshold and set an unvalid index -1
+    num_warn = np.sum(dist * EARTH_RADIUS_KM > threshold)
+    if num_warn:
+        LOGGER.warning('Distance to closest centroid is greater than %s'
+                       'km for %s coordinates.', threshold, num_warn)
+        assigned[dist * EARTH_RADIUS_KM > threshold] = -1
+
+    if check_antimeridian:
+        assigned = index_antimeridian_strip(np.radians(centroids),
+                                            np.radians(coordinates[idx]),
```

Comment:
Why do you convert everything to radians? If you really need the conversion, you can still do it within the function `index_antimeridian_strip`. I suggest to stick to degrees for compatibility with the other functions in this module.

Reply:
Yes, it is better indeed.